stdlib/public/SDK/Foundation/Foundation.swift

```diff
@@ -352,24 +433,54 @@ extension CGFloat : _ObjectiveCBridgeable {
 }
 
 // Literal support for NSNumber
-extension NSNumber : ExpressibleByFloatLiteral, ExpressibleByIntegerLiteral,
-                     ExpressibleByBooleanLiteral {
+extension NSNumber
+  : ExpressibleByFloatLiteral,
+    ExpressibleByIntegerLiteral,
+    ExpressibleByBooleanLiteral
+{
   /// Create an instance initialized to `value`.
   public required convenience init(integerLiteral value: Int) {
+    // FIXME(id-as-any): how to replace self with a type-preserving NSNumber?
```

Comment:
I think these are fine. You're making an NSNumber, therefore it can have fluid type.